lib/puma/cluster.rb

```diff
@@ -218,14 +218,9 @@ def check_workers(force=false)
       # during this loop by giving the kernel time to kill them.
       sleep 1 if any
 
-      while @workers.any?
-        pid = Process.waitpid(-1, Process::WNOHANG)
-        break unless pid
```

Comment:
@MSP-Greg talking about this break here. also the waitpid is slightly different. wondering what effect the differences will have.

Reply:
1. `Process.waitpid(-1, Process::WNOHANG)` will check **all** child processes, not just workers.  Not sure if that is an issue or not.

2. With the '-1' parameter, I believe the existing code will return pid's for all child processes that are shut down, and then stop from the break.  I believe that the following will do the same, just limited to workers (and a bit cleaned up):
```ruby
@workers.reject! { |w| Process.waitpid(w.pid, Process::WNOHANG) }
```

Locally, I'm just windows, so the only way I can check code using `fork` is thru Travis.  That also leaves me somewhat unfamiliar with it, so I'll consider test results, which are green.  I'm also aware that workers may be shutdown due to a misbehaving application, and that can be difficult to test.

Hence, I'll defer to your judgement...

Reply:
> Process.waitpid(-1, Process::WNOHANG) will check all child processes, not just workers. Not sure if that is an issue or not.

I think this will be fine unless anyone is doing really weird stuff with `fork`.